MachineLearningPrototype/dataCollection/collect_data.py

```diff
@@ -55,15 +55,34 @@ def store_issue_details(issue, repo, db):
 	issue_details['created_at'] = issue['created_at']
 	issue_details['updated_at'] = issue['updated_at']
 	issue_details['labels'] = issue['labels']
-	
+
 	issue_details['issue_content_path'] = issue_content_path
 	# issue_details['test_output_path'] = test_output_path
 	issue_details['issue_content_quotation_path'] = issue_content_quotation_path
+	trss_jenkins_output_path = f'./data/GitHubData/JenkinsOutput/{issue_uid}'
 	#Store issue content at issue_content_path
 	store_on_fs(issue['body'], issue_content_path)
 	store_quotation_result = store_on_fs(extract_quotation_content(issue['body']), issue_content_quotation_path)
 	if store_quotation_result:
 		issue_details['issue_content_quotation_path'] = issue_content_quotation_path
+
+	content_extracted = extract_jenkins_link_and_testname(issue['body'])
+	issue_quoted_jenkins_links = content_extracted[0]
+	issue_quoted_test_names = content_extracted[1]
+	if issue_quoted_jenkins_links != []:
+    	issue_details['issue_quoted_jenkins_links'] = issue_quoted_jenkins_links
+	if issue_quoted_test_names != []:
+    	issue_details['issue_quoted_test_names'] = issue_quoted_test_names
+
+	if issue_quoted_jenkins_links != [] and issue_quoted_test_names != []:
+        for link in issue_quoted_jenkins_links:
+            for testname in issue_quoted_test_names:
+                trss_jenkins_output = query_trss_for_jenkins_output(link, testname)
+				if trss_jenkins_output:
+					store_trss_jenkins_output = store_on_fs(trss_jenkins_output, trss_jenkins_output_path)
+					issue_details['trss_jenkins_output_path'] = trss_jenkins_output_path
```

Comment:
If we have multiple tests from multiple Jenkins, the current logic is to write them into the same file. Is this expected?

Reply:
@llxia the problem is I'm not getting how this thing is gonna work, after running the code only one file was saved in the newly made `JenkinsOutput` folder. And I'm trying to follow the instructions like what the issue wants and following previous comments in this PR

Reply:
the path `trss_jenkins_output_path` has been made to be `./data/GitHubData/JenkinsOutput/{issue_uid}`, so I think it should read one by one and proecss it and store it in individual files.

Reply:
Current code will only store the last jenkins output queried (with `"w"` in `store_on_fs`), but we need to save all related jenkins output, either in a combined file (or in multiple files). 
In addition, since Jenkins output expires very soon, if a issue is only updated, we need to make sure the less jenkins output will not overwrite the original one, as mentioned in Issue https://github.com/adoptium/aqa-test-tools/issues/486. Possibly we can compare new `issue_quoted_jenkins_links` and `issue_quoted_test_names` with the ones in DB, then query TRSS for only newly added ones, then `attach` new jenkins output to the original file.